db-client-java/src/main/java/com/eventstore/dbclient/GrpcClient.java

```diff
@@ -136,37 +141,43 @@ private boolean discover(UUID previousId, Optional<Endpoint> candidate) {
         long attempts = 1;
 
         // It means we already created a new channel and it was old request.
-        if (!currentChannelId.equals(previousId))
+        if (!currentChannelId.equals(previousId)) {
+            logger.debug("Skipping connection attempt as new connection to endpoint [{}] has already been created.", endpoint);
```

Comment:
This might also be better on TRACE level.

Reply:
How many times would we expect to see this log per discovery? If it would just be an extra one or two lines in the log then leaving it as DEBUG would be fine in my opinion

Reply:
It depends on the load on the client, i.e. rate of executed operations. Usually it should only happen 2-3 times per failover.